mmv1/third_party/terraform/services/resourcemanager/resource_google_project_migrate.go

```diff
@@ -24,11 +24,31 @@ func resourceGoogleProjectMigrateState(v int, s *terraform.InstanceState, meta i
 			return s, err
 		}
 		return s, nil
+	case 1:
+		log.Println("[INFO] Found Google Project State v1; migrating to v2")
+		s, err := migrateGoogleProjectStateV1toV2(s, meta.(*transport_tpg.Config))
+		if err != nil {
+			return s, err
+		}
+		return s, nil
 	default:
 		return s, fmt.Errorf("Unexpected schema version: %d", v)
 	}
 }
 
+// This migration adjusts google_project resources to replace skip_delete with deletion_policy.
+func migrateGoogleProjectStateV1toV2(s *terraform.InstanceState, config *transport_tpg.Config) (*terraform.InstanceState, error) {
+	log.Printf("[DEBUG] Attributes before migration: %#v", s.Attributes)
+
+	if s.Attributes["skip_delete"] == "true" {
+		s.Attributes["deletion_policy"] = "ABANDON"
+		s.Attributes["skip_delete"] = "false"
```

Comment:
@zli82016 I'm not sure what's the guidance here.... Do we leave skip_delete as it is or explicitly set to false to actually simulate behavior when the field will be deprecated?

Reply:
After talking with our team, the state upgrader is not necessary because of the default value of `deletion_policy`. Sorry for the extra work. Can you remove the state upgrader? Thanks.

Reply:
Sure, removed state upgrader.

Reply:
`StateUpgraders` should be added instead of modifying `MigrateState`, as `MigrateState` is deprecated. 

This is the [doc](https://developer.hashicorp.com/terraform/plugin/sdkv2/resources/state-migration#terraform-v0-12-sdk-state-migrations) how to add `StateUpgraders`

Reply:
We only need to migrate the state when `skip_delete` is not present in state.

Reply:
Added.